### ISSUE TITLE
Batching doucments 

### ISSUE THREAD

**Issue by sajastu** [MaartenGr/KeyBERT#79]
Hi,

I'm wondering if it would be possible to batch a set of documents (like 10) in parallel and then pass it through the KeyBert model? Currently, the example in the documentation is only one single document. Also, there's one other open issue that is similar to this (multiple-document), but I was not convinced how I can use this feature if it exists. 

Best,

**Comment by MaartenGr** [comment]
There are two ways of processing documents. 

The first is processing a single document at a time. Doing so allows you to fine-tune the resulting candidate keywords with MMR or MaxSumSim. Especially those methods are helpful in creating interesting and diverse keywords. 

The second is processing multiple documents at a time. When you pass a list of documents to KeyBERT, the following happens. First, it extracts all unique words from all documents and creates embeddings from those words. This speeds up processing as each unique word only needs to be embedded once. However, MMR and MaxSumSim are not implemented as they cannot be processed in bulk. Moreover, parallelizing the extraction of embeddings is likely to come with a number of issues since creating the embeddings is typically done using a GPU. 

In other words, I would advise sticking with processing a single document at a time.

**Comment by sajastu** [comment]
It makes sense. I could manage to perform my task with single document processing at a time. Closing this issue now...
